packages/components/src/radio-control/index.tsx

```diff
@@ -104,4 +77,31 @@ export function RadioControl(
 	);
 }
 
+/**
+ * Render a user interface to select the user type using radio inputs.
+ *
+ * ```jsx
+ * import { RadioControl } from '@wordpress/components';
+ * import { useState } from '@wordpress/element';
+ *
+ * const MyRadioControl = () => {
+ *   const [ option, setOption ] = useState( 'a' );
+ *
+ *   return (
+ *     <RadioControl
+ *       label="User type"
+ *       help="The type of the current user"
+ *       selected={ option }
+ *       options={ [
+ *         { label: 'Author', value: 'a' },
+ *         { label: 'Editor', value: 'e' },
+ *       ] }
+ *       onChange={ ( value ) => setOption( value ) }
+ *     />
+ *   );
+ * };
+ * ```
+ */
+export const RadioControl = forwardRef( UnforwardedRadioControl );
```

Comment:
@ciampo Does that function name seem confusing? UnforwardedRadioControl? Not sure if other components use that but it just seems weird especially reading from start to finish as the function now forwards the ref.

Reply:
This is pattern that @mirka and I have adopted in the context of the TypeScript migration that we're undertaking across the components in the package (see [this guide](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md#refactoring-a-component-to-typescript) for more details)

In order to have Storybook extract and show prop types correctly, we need to have a named export for the component ("RadioControl" in this case). Therefore, we had to find a suitable name for the intermediate representation of the component, and we opted for:

- adding the "unforwarded" prefix when the component is later wrapped in a `forwardRef` call
- adding the "unconnected" prefix when the component is later wrapped in a `contextConnect` call

Hope that's a bit more clear!